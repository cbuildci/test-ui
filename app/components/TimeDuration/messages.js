/*
 * TimeDuration Messages
 *
 * This contains all the text for the TimeDuration component.
 */

import { defineMessages } from 'react-intl';

export default defineMessages({
    duration: {
        id: 'app.components.TimeDuration.duration',
        defaultMessage:
            `{value, number} {unit, select,
                seconds {{value, plural, one {second} other {seconds} }}
                minutes {{value, plural, one {minute} other {minutes} }}
                hours {{value, plural, one {hour} other {hours} }}
                days {{value, plural, one {day} other {days} }}
                weeks {{value, plural, one {week} other {weeks} }}
                months {{value, plural, one {month} other {months} }}
                years {{value, plural, one {year} other {years} }}
            }`,
    },
});
