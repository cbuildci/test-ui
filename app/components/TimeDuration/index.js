/**
 * TimeDuration
 */

import React from 'react';
import PropTypes from 'prop-types';

import { FormattedMessage } from 'react-intl';
import messages from './messages';

function TimeDuration({
    seconds,
}) {
    let unit = 'seconds';
    let value = seconds;

    if (value >= 31536000) {
        unit = 'years';
        value = Math.round(value / 3153600) / 10;
    }
    else if (value >= 3942000) {
        unit = 'months';
        value = Math.round(value / 394200) / 10;
    }
    else if (value >= 604800) {
        unit = 'weeks';
        value = Math.round(value / 60480) / 10;
    }
    else if (value >= 86400) {
        unit = 'days';
        value = Math.round(value / 8640) / 10;
    }
    else if (value >= 3600) {
        unit = 'hours';
        value = Math.round(value / 360) / 10;
    }
    else if (value >= 60) {
        unit = 'minutes';
        value = Math.round(value / 6) / 10;
    }

    return (
        <FormattedMessage
            {...messages.duration}
            values={{ value, unit }}
        />
    );
}

TimeDuration.propTypes = {
    seconds: PropTypes.number.isRequired,
};

export default TimeDuration;
