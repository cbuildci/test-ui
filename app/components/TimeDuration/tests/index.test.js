import React from 'react';
import { snapshots } from '../../../../internals/testing/snapshot-util';
import TimeDuration from '../index';

describe('<TimeDuration />', () => {
    it('Should render expected JSX', () => {
        snapshots(
            <TimeDuration seconds={0}/>
        );

        snapshots(
            <TimeDuration seconds={59}/>
        );

        // 1 Minute
        snapshots(
            <TimeDuration seconds={60}/>
        );

        snapshots(
            <TimeDuration seconds={3599}/>
        );

        // 1 Hour
        snapshots(
            <TimeDuration seconds={3600}/>
        );

        snapshots(
            <TimeDuration seconds={86399}/>
        );

        // 1 Day
        snapshots(
            <TimeDuration seconds={86400}/>
        );

        snapshots(
            <TimeDuration seconds={604799}/>
        );

        // 1 Week
        snapshots(
            <TimeDuration seconds={604800}/>
        );

        snapshots(
            <TimeDuration seconds={3941999}/>
        );

        // ~1.5 Months (365 / 12 * 1.5)
        snapshots(
            <TimeDuration seconds={3942000}/>
        );

        // ~5 Months
        snapshots(
            <TimeDuration seconds={19710000}/>
        );

        snapshots(
            <TimeDuration seconds={31535999}/>
        );

        // 1 Year
        snapshots(
            <TimeDuration seconds={31536000}/>
        );

        // 10 Years
        snapshots(
            <TimeDuration seconds={315360000}/>
        );
    });
});
