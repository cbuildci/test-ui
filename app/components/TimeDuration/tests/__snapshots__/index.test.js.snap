// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TimeDuration /> Should render expected JSX 1`] = `
<FormattedMessage
  defaultMessage="{value, number} {unit, select,
                seconds {{value, plural, one {second} other {seconds} }}
                minutes {{value, plural, one {minute} other {minutes} }}
                hours {{value, plural, one {hour} other {hours} }}
                days {{value, plural, one {day} other {days} }}
                weeks {{value, plural, one {week} other {weeks} }}
                months {{value, plural, one {month} other {months} }}
                years {{value, plural, one {year} other {years} }}
            }"
  id="app.components.TimeDuration.duration"
  values={
    Object {
      "unit": "seconds",
      "value": 0,
    }
  }
/>
`;

exports[`<TimeDuration /> Should render expected JSX 2`] = `
<span>
  0 seconds
</span>
`;

exports[`<TimeDuration /> Should render expected JSX 3`] = `
<FormattedMessage
  defaultMessage="{value, number} {unit, select,
                seconds {{value, plural, one {second} other {seconds} }}
                minutes {{value, plural, one {minute} other {minutes} }}
                hours {{value, plural, one {hour} other {hours} }}
                days {{value, plural, one {day} other {days} }}
                weeks {{value, plural, one {week} other {weeks} }}
                months {{value, plural, one {month} other {months} }}
                years {{value, plural, one {year} other {years} }}
            }"
  id="app.components.TimeDuration.duration"
  values={
    Object {
      "unit": "seconds",
      "value": 59,
    }
  }
/>
`;

exports[`<TimeDuration /> Should render expected JSX 4`] = `
<span>
  59 seconds
</span>
`;

exports[`<TimeDuration /> Should render expected JSX 5`] = `
<FormattedMessage
  defaultMessage="{value, number} {unit, select,
                seconds {{value, plural, one {second} other {seconds} }}
                minutes {{value, plural, one {minute} other {minutes} }}
                hours {{value, plural, one {hour} other {hours} }}
                days {{value, plural, one {day} other {days} }}
                weeks {{value, plural, one {week} other {weeks} }}
                months {{value, plural, one {month} other {months} }}
                years {{value, plural, one {year} other {years} }}
            }"
  id="app.components.TimeDuration.duration"
  values={
    Object {
      "unit": "minutes",
      "value": 1,
    }
  }
/>
`;

exports[`<TimeDuration /> Should render expected JSX 6`] = `
<span>
  1 minute
</span>
`;

exports[`<TimeDuration /> Should render expected JSX 7`] = `
<FormattedMessage
  defaultMessage="{value, number} {unit, select,
                seconds {{value, plural, one {second} other {seconds} }}
                minutes {{value, plural, one {minute} other {minutes} }}
                hours {{value, plural, one {hour} other {hours} }}
                days {{value, plural, one {day} other {days} }}
                weeks {{value, plural, one {week} other {weeks} }}
                months {{value, plural, one {month} other {months} }}
                years {{value, plural, one {year} other {years} }}
            }"
  id="app.components.TimeDuration.duration"
  values={
    Object {
      "unit": "minutes",
      "value": 60,
    }
  }
/>
`;

exports[`<TimeDuration /> Should render expected JSX 8`] = `
<span>
  60 minutes
</span>
`;

exports[`<TimeDuration /> Should render expected JSX 9`] = `
<FormattedMessage
  defaultMessage="{value, number} {unit, select,
                seconds {{value, plural, one {second} other {seconds} }}
                minutes {{value, plural, one {minute} other {minutes} }}
                hours {{value, plural, one {hour} other {hours} }}
                days {{value, plural, one {day} other {days} }}
                weeks {{value, plural, one {week} other {weeks} }}
                months {{value, plural, one {month} other {months} }}
                years {{value, plural, one {year} other {years} }}
            }"
  id="app.components.TimeDuration.duration"
  values={
    Object {
      "unit": "hours",
      "value": 1,
    }
  }
/>
`;

exports[`<TimeDuration /> Should render expected JSX 10`] = `
<span>
  1 hour
</span>
`;

exports[`<TimeDuration /> Should render expected JSX 11`] = `
<FormattedMessage
  defaultMessage="{value, number} {unit, select,
                seconds {{value, plural, one {second} other {seconds} }}
                minutes {{value, plural, one {minute} other {minutes} }}
                hours {{value, plural, one {hour} other {hours} }}
                days {{value, plural, one {day} other {days} }}
                weeks {{value, plural, one {week} other {weeks} }}
                months {{value, plural, one {month} other {months} }}
                years {{value, plural, one {year} other {years} }}
            }"
  id="app.components.TimeDuration.duration"
  values={
    Object {
      "unit": "hours",
      "value": 24,
    }
  }
/>
`;

exports[`<TimeDuration /> Should render expected JSX 12`] = `
<span>
  24 hours
</span>
`;

exports[`<TimeDuration /> Should render expected JSX 13`] = `
<FormattedMessage
  defaultMessage="{value, number} {unit, select,
                seconds {{value, plural, one {second} other {seconds} }}
                minutes {{value, plural, one {minute} other {minutes} }}
                hours {{value, plural, one {hour} other {hours} }}
                days {{value, plural, one {day} other {days} }}
                weeks {{value, plural, one {week} other {weeks} }}
                months {{value, plural, one {month} other {months} }}
                years {{value, plural, one {year} other {years} }}
            }"
  id="app.components.TimeDuration.duration"
  values={
    Object {
      "unit": "days",
      "value": 1,
    }
  }
/>
`;

exports[`<TimeDuration /> Should render expected JSX 14`] = `
<span>
  1 day
</span>
`;

exports[`<TimeDuration /> Should render expected JSX 15`] = `
<FormattedMessage
  defaultMessage="{value, number} {unit, select,
                seconds {{value, plural, one {second} other {seconds} }}
                minutes {{value, plural, one {minute} other {minutes} }}
                hours {{value, plural, one {hour} other {hours} }}
                days {{value, plural, one {day} other {days} }}
                weeks {{value, plural, one {week} other {weeks} }}
                months {{value, plural, one {month} other {months} }}
                years {{value, plural, one {year} other {years} }}
            }"
  id="app.components.TimeDuration.duration"
  values={
    Object {
      "unit": "days",
      "value": 7,
    }
  }
/>
`;

exports[`<TimeDuration /> Should render expected JSX 16`] = `
<span>
  7 days
</span>
`;

exports[`<TimeDuration /> Should render expected JSX 17`] = `
<FormattedMessage
  defaultMessage="{value, number} {unit, select,
                seconds {{value, plural, one {second} other {seconds} }}
                minutes {{value, plural, one {minute} other {minutes} }}
                hours {{value, plural, one {hour} other {hours} }}
                days {{value, plural, one {day} other {days} }}
                weeks {{value, plural, one {week} other {weeks} }}
                months {{value, plural, one {month} other {months} }}
                years {{value, plural, one {year} other {years} }}
            }"
  id="app.components.TimeDuration.duration"
  values={
    Object {
      "unit": "weeks",
      "value": 1,
    }
  }
/>
`;

exports[`<TimeDuration /> Should render expected JSX 18`] = `
<span>
  1 week
</span>
`;

exports[`<TimeDuration /> Should render expected JSX 19`] = `
<FormattedMessage
  defaultMessage="{value, number} {unit, select,
                seconds {{value, plural, one {second} other {seconds} }}
                minutes {{value, plural, one {minute} other {minutes} }}
                hours {{value, plural, one {hour} other {hours} }}
                days {{value, plural, one {day} other {days} }}
                weeks {{value, plural, one {week} other {weeks} }}
                months {{value, plural, one {month} other {months} }}
                years {{value, plural, one {year} other {years} }}
            }"
  id="app.components.TimeDuration.duration"
  values={
    Object {
      "unit": "weeks",
      "value": 6.5,
    }
  }
/>
`;

exports[`<TimeDuration /> Should render expected JSX 20`] = `
<span>
  6.5 weeks
</span>
`;

exports[`<TimeDuration /> Should render expected JSX 21`] = `
<FormattedMessage
  defaultMessage="{value, number} {unit, select,
                seconds {{value, plural, one {second} other {seconds} }}
                minutes {{value, plural, one {minute} other {minutes} }}
                hours {{value, plural, one {hour} other {hours} }}
                days {{value, plural, one {day} other {days} }}
                weeks {{value, plural, one {week} other {weeks} }}
                months {{value, plural, one {month} other {months} }}
                years {{value, plural, one {year} other {years} }}
            }"
  id="app.components.TimeDuration.duration"
  values={
    Object {
      "unit": "months",
      "value": 1,
    }
  }
/>
`;

exports[`<TimeDuration /> Should render expected JSX 22`] = `
<span>
  1 month
</span>
`;

exports[`<TimeDuration /> Should render expected JSX 23`] = `
<FormattedMessage
  defaultMessage="{value, number} {unit, select,
                seconds {{value, plural, one {second} other {seconds} }}
                minutes {{value, plural, one {minute} other {minutes} }}
                hours {{value, plural, one {hour} other {hours} }}
                days {{value, plural, one {day} other {days} }}
                weeks {{value, plural, one {week} other {weeks} }}
                months {{value, plural, one {month} other {months} }}
                years {{value, plural, one {year} other {years} }}
            }"
  id="app.components.TimeDuration.duration"
  values={
    Object {
      "unit": "months",
      "value": 5,
    }
  }
/>
`;

exports[`<TimeDuration /> Should render expected JSX 24`] = `
<span>
  5 months
</span>
`;

exports[`<TimeDuration /> Should render expected JSX 25`] = `
<FormattedMessage
  defaultMessage="{value, number} {unit, select,
                seconds {{value, plural, one {second} other {seconds} }}
                minutes {{value, plural, one {minute} other {minutes} }}
                hours {{value, plural, one {hour} other {hours} }}
                days {{value, plural, one {day} other {days} }}
                weeks {{value, plural, one {week} other {weeks} }}
                months {{value, plural, one {month} other {months} }}
                years {{value, plural, one {year} other {years} }}
            }"
  id="app.components.TimeDuration.duration"
  values={
    Object {
      "unit": "months",
      "value": 8,
    }
  }
/>
`;

exports[`<TimeDuration /> Should render expected JSX 26`] = `
<span>
  8 months
</span>
`;

exports[`<TimeDuration /> Should render expected JSX 27`] = `
<FormattedMessage
  defaultMessage="{value, number} {unit, select,
                seconds {{value, plural, one {second} other {seconds} }}
                minutes {{value, plural, one {minute} other {minutes} }}
                hours {{value, plural, one {hour} other {hours} }}
                days {{value, plural, one {day} other {days} }}
                weeks {{value, plural, one {week} other {weeks} }}
                months {{value, plural, one {month} other {months} }}
                years {{value, plural, one {year} other {years} }}
            }"
  id="app.components.TimeDuration.duration"
  values={
    Object {
      "unit": "years",
      "value": 1,
    }
  }
/>
`;

exports[`<TimeDuration /> Should render expected JSX 28`] = `
<span>
  1 year
</span>
`;

exports[`<TimeDuration /> Should render expected JSX 29`] = `
<FormattedMessage
  defaultMessage="{value, number} {unit, select,
                seconds {{value, plural, one {second} other {seconds} }}
                minutes {{value, plural, one {minute} other {minutes} }}
                hours {{value, plural, one {hour} other {hours} }}
                days {{value, plural, one {day} other {days} }}
                weeks {{value, plural, one {week} other {weeks} }}
                months {{value, plural, one {month} other {months} }}
                years {{value, plural, one {year} other {years} }}
            }"
  id="app.components.TimeDuration.duration"
  values={
    Object {
      "unit": "years",
      "value": 10,
    }
  }
/>
`;

exports[`<TimeDuration /> Should render expected JSX 30`] = `
<span>
  10 years
</span>
`;
